src/genome.py

```diff
@@ -187,8 +272,41 @@ def insert_te(self, pos: int, length: int) -> int:
 
         Returns a new ID for the transposable element.
         """
-        ...  # FIXME
-        return -1
+        self.TE_counter += 1
+
+        # disable existing TE. If it is active it is continuous. 
+        for te in self.TEs:
+            if self.TEs[te][2] == 'A':    
+                start = self.TEs[te][0]
+                end = self.TEs[te][0] + self.TEs[te][1]
+                if start < pos and pos < end:         
+                    self.disable_te(te)
+                    break
+        
+        # update genome with insertion
+        count = 1
+
+        current = self.head.next
+        # iterator
+        while current is not self.head:
+            if count == pos:
+                # insert 'A'*length at pos
+                for _ in range(length):
+                    insert_after(current, 'A')    # or insert before? no
+                break
+            current = current.next
+            count += 1
+        
+        # update pos for every TE positioned after the insertion
+        for te in self.TEs:
+            start = self.TEs[te][0]
+            if pos <= start:
+                self.TEs[te][0] += length
+
```

Comment:
Ok, but it is a little expensive. Do you really need to know the exact position of each TE?

Reply:
I use the position to know when I hit an existing TE. Can't think of another way.

Reply:
You can store the TE ID in the nucleotide. Then you know if you are looking at a TE at any given position.